python/run_plotXSTemplates.py

```diff
@@ -0,0 +1,413 @@
+    CMSPrelim.SetTextFont(42);
+    CMSPrelim.SetTextAlign(31); #// align right
+ #   //CMSPrelim.DrawLatex(0.93, 0.96,"36.8 fb^{-1} at #sqrt{s} = 13 TeV");
+    #CMSPrelim.DrawLatex(0.93, 0.96,"58.9 fb^{-1} at #sqrt{s} = 13 TeV");
+    CMSPrelim.DrawLatex(0.93, 0.96,str(lumi[opt.YEAR])+ " fb^{-1} at #sqrt{s} = 13 TeV");
+
+    CMSPrelim.SetTextSize(0.9*c.GetTopMargin());
+    CMSPrelim.SetTextFont(62);
+    CMSPrelim.SetTextAlign(11); #// align right
+    CMSPrelim.DrawLatex(0.27, 0.85, "CMS");
+
+    CMSPrelim.SetTextSize(0.7*c.GetTopMargin());
+    CMSPrelim.SetTextFont(52);
+    CMSPrelim.SetTextAlign(11);
+    CMSPrelim.DrawLatex(0.25, 0.8, "Preliminary");
+
+
+def setLegendProperties(leg, sHeader = "skip", fillStyle=0, fillColor=0):
+    if not leg: return
+    if (sHeader!="skip"): leg.SetHeader(sHeader)
+    leg.SetFillColor(0)
+    leg.SetFillStyle(0)
+    leg.SetBorderSize(0)
+    leg.SetTextFont(43)
+
+
+def normaliseHist(h1D, norm = 1.):
+    if (h1D.Integral()==0): return -1; 
+    h1D.Scale(norm/h1D.Integral());
+    return 0;
+
+
+def fillEmptyBinsHist1D(h1D, floor):
+    nXbins=h1D.GetNbinsX();
+    #for(int i=1, i<=nXbins, i++):
+#    for(int i=1, i<=nXbins, i++):
+    for i in range(nXbins):
+        #h1D.SetBinContent(i,h1D.GetBinContent(i)+floor);
+        h1D.SetBinContent(i+1,h1D.GetBinContent(i+1)+floor);
+    return 0;
+
+
+
+def smoothAndNormaliseTemplate1D(h1D, norm = 1.):
+    # smooth
+    h1D.Smooth();
+    #h1D.Smooth(10000);
+    #norm + floor + norm
+    normaliseHist(h1D, norm);
+    fillEmptyBinsHist1D(h1D,.001/(h1D.GetNbinsX()));
+    normaliseHist(h1D, norm);
+ 
+def normaliseHist1D(h1D, norm = 1.):
+    if (h1D.Integral()==0):
+	 return -1; 
+    h1D.Scale(norm/h1D.Integral());
+
+    return 0;
+
+
+#act_bins_yml, doubleDiff = binning(opt.OBSNAME)
+
+obsTag = opt.OBSNAME
+
+if OneDOr2DObs==2:
+    binRange = [str(act_bins_yml[i][0])+'_'+str(act_bins_yml[i][1])+'_'+str(act_bins_yml[i][2])+'_'+str(act_bins_yml[i][3]) for i in range(nBins)]
+    obsTag = obsTag.split(' vs ')[0]+'_'+obsTag.split(' vs ')[1]
+    binRangeLeg = [str(act_bins_yml[i][0])+'<'+obsTag+'<'+str(act_bins_yml[i][1])+'/'+str(act_bins_yml[i][2])+'<'+obsTag+'<'+str(act_bins_yml[i][3]) for i in range(nBins)]
+else:
+    binRange = [str(act_bins_yml[i])+'_'+str(act_bins_yml[i+1]) for i in range(nBins)]
+    binRangeLeg = [str(act_bins_yml[i])+'<'+obsTag+'<'+str(act_bins_yml[i+1]) for i in range(nBins)]
+
+print(binRange)
+print(binRangeLeg)
+
+
+setTDRStyle()
+c1 = ROOT.TCanvas('c1',"",0,0,800,800)
+c1.cd(1)
+c1.SetLogy(0)
+gStyle.SetOptStat('')
+gStyle.SetPalette(1)
+c1.GetPad(0).SetRightMargin(0.05)
+c1.GetPad(0).SetLeftMargin(0.15)
+c1.GetPad(0).SetTopMargin(0.05)
+c1.GetPad(0).SetBottomMargin(0.15)
+
+bkgName=['qqZZ','ggZZ','ZJetsCR']
+sTemplateDirName = os.getcwd()+"/templates/templatesXS_"+str(year)+"/DTreeXS_"+obsTag+"/13TeV/";
+sPlotsStore = os.getcwd()+"/templates/plotsXS_"+str(year)+"/"+obsTag+"/";
```

Comment:
Just checked your macro. This directory does not exist to its unable to same the plots. Fix it. 

You can use the already defined function `GetDirectory` in the Utils module. Here,

https://github.com/ram1123/Fiducial_XS/blob/844721c58f073cbd58f6a2c731382206fa53f694/python/Utils.py#L125-L132